src/pudl/transform/phmsagas.py

```diff
@@ -179,16 +179,113 @@ def core_phmsagas__yearly_distribution_operators(
     # Standardize telephone and fax number format and drop (000)-000-0000
     df = standardize_phone_column(df, ["preparer_phone", "preparer_fax"])
 
+    # Drop duplicates
+    df = df.drop_duplicates()
+
+    # Identify non-unique groups based on our PKs
+    non_unique_groups = df[
+        df.groupby(["operator_id_phmsa", "report_number"])["report_number"].transform(
+            "size"
+        )
+        > 1
+    ]
+
+    # Apply some custom filtering logic to non-unique groups
+    filtered_non_unique_rows = non_unique_groups.groupby(
+        ["operator_id_phmsa", "report_number"], group_keys=False
+    ).apply(combined_filter)
+
+    # Combine filtered non-unique rows with untouched unique rows
+    unique_rows = df.drop(non_unique_groups.index)
+    df = pd.concat([unique_rows, filtered_non_unique_rows], ignore_index=True)
+
     return df
 
 
+def filter_if_test_in_address(group: pd.DataFrame) -> pd.DataFrame:
+    """Filters out rows with "test" in address columns. The logic is as follows:
+
+    1. For any group of rows with the same combination of "operator_id_phmsa"
+       and "report_number":
+        - If at least one row in the group does not contain the string "test"
+          (case-insensitive) in either "office_address_street" or
+          "headquarters_address_street", keep only the rows in the group
+          that do not contain "test" in these columns.
+        - If all rows in the group contain "test" in either of the columns,
+          leave the group unchanged.
```

Comment:
The indentation here is causing `make docs-build` to fail. Should look like:

```suggestion
    1. For any group of rows with the same combination of "operator_id_phmsa"
       and "report_number":
       - If at least one row in the group does not contain the string "test"
       (case-insensitive) in either "office_address_street" or
       "headquarters_address_street", keep only the rows in the group
       that do not contain "test" in these columns.
       - If all rows in the group contain "test" in either of the columns,
       leave the group unchanged.
```